corehq/apps/smsbillables/models.py

```diff
@@ -27,23 +28,47 @@ class SmsGatewayFeeCriteria(models.Model):
     backend_instance = models.CharField(max_length=255, db_index=True, null=True)
     direction = models.CharField(max_length=10, db_index=True, choices=DIRECTION_CHOICES)
     country_code = models.IntegerField(max_length=5, null=True, blank=True, db_index=True)
+    prefix = models.CharField(max_length=10, blank=True, default="", db_index=True)
 
     @classmethod
-    def get_most_specific(cls, backend_api_id, direction, backend_instance=None, country_code=None):
+    def get_most_specific(cls, backend_api_id, direction,
+                          backend_instance=None, country_code=None, national_number=None):
         """
         Gets the most specific criteria available based on (and in order of preference for optional):
         - backend_api_id
         - direction
         - backend_instance (optional)
-        - country_code (optional)
+        - country_code and prefix (optional)
         """
         all_possible_criteria = cls.objects.filter(backend_api_id=backend_api_id, direction=direction)
 
         if all_possible_criteria.count() == 0:
             return None
 
+        national_number = str(national_number) if national_number is not None else ""
+        if national_number:
+            national_number = '0' * (10 - len(national_number)) + national_number
```

Comment:
I see - but so then isn't this code assuming that the length of the national number is always 10 (I don't think that's a safe assumption to make)?